### PR TITLE
fix(av-moderation) Advanced moderation improvements

### PR DESCRIPTION
**Pull request by robertpin**

### CODE REVIEW

File: react/features/participants-pane/components/web/MeetingParticipantItem.js [hunk 1]
```diff
@@ -155,12 +169,41 @@ function MeetingParticipantItem({
     participantActionEllipsisLabel,
     youText
 }: Props) {
+
+    const [ hasAudioLevels, setHasAudioLevel ] = useState(false);
+    const [ registeredEvent, setRegisteredEvent ] = useState(false);
+
+    const _updateAudioLevel = level => {
```
Comment (vp8x8) [inline react/features/participants-pane/components/web/MeetingParticipantItem.js:176]: This function will be recreated on every render. Maybe it's a good idea to use `useCallback`.

**Review by horymury — CHANGES_REQUESTED** [review]

File: react/features/participants-pane/components/web/MeetingParticipantItem.js [hunk 2]
```diff
@@ -173,34 +173,44 @@ function MeetingParticipantItem({
...
+            if (_audioTrack && registeredEvent) {
+                const { jitsiTrack } = _audioTrack;
+
+                jitsiTrack && jitsiTrack.off(JitsiTrackEvents.TRACK_AUDIO_LEVEL_CHANGED, _updateAudioLevel);
+            }
+        };
     }, [ _audioTrack ]);
 
-    const _getAudioMediaState = useMemo(() => {
+    const _getAudioMediaState = () => {
```
Comment (vp8x8) [inline react/features/participants-pane/components/web/MeetingParticipantItem.js:202]: Do we need a function for this? Wouldn't a simple variable suffice?

Comment (robertpin) [reply react/features/participants-pane/components/web/MeetingParticipantItem.js]: Yes, it can be a variable. Changed

**Comment by vp8x8** [comment]
@horymury this looks good from my point of view. Please merge when/if you think appropriate.

**Comment by robertpin** [comment]
Jenkins test this please.

File: react/features/av-moderation/middleware.js [hunk 3]
```diff
@@ -35,21 +37,31 @@ import {
...
     isParticipantPending
 } from './functions';
+import { ASKED_TO_UNMUTE_FILE } from './sounds';
 
-const VIDEO_MODERATION_NOTIFICATION_ID = 'video-moderation';
-const AUDIO_MODERATION_NOTIFICATION_ID = 'audio-moderation';
-const CS_MODERATION_NOTIFICATION_ID = 'video-moderation';
+export const AUDIO_MODERATION_NOTIFICATION_ID = 'audio-moderation';
+export const VIDEO_MODERATION_NOTIFICATION_ID = 'video-moderation';
+export const CS_MODERATION_NOTIFICATION_ID = 'screensharing-moderation';
```
Comment (horymury) [inline react/features/av-moderation/middleware.js:50]: middleware sounds like a weird place to define those. Maybe define them privately on the constants.js?

**Review by horymury — APPROVED** [review]

**Review by horymury — DISMISSED** [review]

**Review by horymury — APPROVED** [review]

**Pull request merged**